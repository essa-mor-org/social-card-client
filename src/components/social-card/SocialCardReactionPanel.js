import React from 'react';

function SocialCardReactionPanel(){
    return null;
}

export default SocialCardReactionPanel;