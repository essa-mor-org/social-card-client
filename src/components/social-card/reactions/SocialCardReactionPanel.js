import React from 'react';
import PropTypes from 'prop-types';

import SocialCardReactions from './SocialCardReactions';
import SocialCardReact from './SocialCardReact';

function SocialCardReactionPanel({ id, like, likes, shares, postComments, likeCallBack, shareCallBack }) {
    return (<React.Fragment>
        <SocialCardReactions
            likes={likes}
            comments={postComments.length}
            shares={shares}
        />
        <SocialCardReact id={id} like={like} likes={likes} shares={shares} onLike={likeCallBack} onShare={shareCallBack} />
    </React.Fragment>);
}

SocialCardReactionPanel.propTypes = {
    id: PropTypes.number.isRequired,
    like: PropTypes.bool,
    likes: PropTypes.number,
    shares: PropTypes.number,
    postComments: PropTypes.arrayOf(PropTypes.number).isRequired,
    likeCallBack: PropTypes.func.isRequired,
    shareCallBack: PropTypes.func.isRequired
};


export default SocialCardReactionPanel;