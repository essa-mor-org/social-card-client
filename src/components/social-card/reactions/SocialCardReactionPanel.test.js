import React from 'react';
import SocialCardReactionPanel from './SocialCardReactionPanel';
import renderer from 'react-test-renderer';

describe('SocialCardReactionPanel test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<SocialCardReactionPanel
                postComments={[]}
				id={1}
				like={true}
				likeCallBack={jest.fn()}
				shareCallBack={jest.fn()}
				likes={157}
				comments= {2}
				shares= {3} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
