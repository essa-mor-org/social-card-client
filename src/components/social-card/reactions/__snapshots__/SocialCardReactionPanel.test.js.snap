// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCardReactionPanel test renders correctly 1`] = `
Array [
  <div
    className="Social_reaction"
>
    <a
        className="Social_reaction_fixed"
        href="#"
    >
        <span
            className="Social_reaction_like"
        />
    </a>
    <span
        className="Social_reaction_strech"
    >
        <a
            className="Social_reaction_number"
            href="#"
        >
            157
        </a>
    </span>
    <a
        className="Social_reaction_number Social_reaction_fixed"
        href="#"
    >
        3
         Shares
    </a>
</div>,
  <div
    className="Social_react"
>
    <a
        className="Reaction_like_active Reaction_like_text_active Reaction_button"
        href="javascript:void(0);"
        onClick={[Function]}
    >
        Like
    </a>
    <a
        className="Reaction_comment Reaction_button"
        href="javascript:void(0);"
        onClick={undefined}
    >
        Comment
    </a>
    <a
        className="Reaction_share Reaction_button"
        href="javascript:void(0);"
        onClick={[Function]}
    >
        Share
    </a>
</div>,
]
`;
